Source/DafnyCore/Verifier/Translator.ExpressionWellformed.cs

```diff
@@ -712,28 +712,52 @@ private record ReadsCheckDelayer(ExpressionTranslator etran, Function selfCallsA
                 }
               }
               // check that the preconditions for the call hold
-              foreach (AttributedExpression p in e.Function.Req) {
-                Expression precond = Substitute(p.E, e.Receiver, substMap, e.GetTypeArgumentSubstitutions());
-                var (errorMessage, successMessage) = CustomErrorMessage(p.Attributes);
-                foreach (var ss in TrSplitExpr(precond, etran, true, out var splitHappened)) {
-                  if (ss.IsChecked) {
-                    var tok = new NestedToken(GetToken(expr), ss.Tok);
-                    var desc = new PODesc.PreconditionSatisfied(errorMessage, successMessage);
-                    if (wfOptions.AssertKv != null) {
-                      // use the given assert attribute only
-                      builder.Add(Assert(tok, ss.E, desc, wfOptions.AssertKv));
-                    } else {
-                      builder.Add(AssertNS(tok, ss.E, desc));
+              // the check for .reads function must be translated explicitly: their declaration lacks
```

Comment:
`must be translated explicitly` is somewhat ambiguous to me, although I guess that you mean we need code specifically for checking that reads clauses are satisfied.

What does "there declaration lacks an explicit precondition" mean? Are reads functions declared? What would it look like if they had an explicit precondition?